modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx

```diff
@@ -306,13 +308,43 @@ function Filters({
 	};
 
 	const onCreationButtonClick = (filterType: EFilterType) => {
-		const availableFieldsList = fields.filter((item) =>
-			FILTER_TYPES[filterType as EFilterType].availableFieldsFilter(item)
-		);
+		let availableFieldsListLength = 0;
+
+		if (Liferay.FeatureFlags['LPD-25905']) {
+			const availableFfilterTypeFields = JSON.parse(
+				JSON.stringify(fields)
+			);
+
+			visit(availableFfilterTypeFields, (field: IFieldTreeItem) => {
+				if (
+					!FILTER_TYPES[
+						filterType as EFilterType
+					].availableFieldsFilter(field)
+				) {
+					field.disabled = true;
```

Comment:
Use `disabled` to extend `IField` as it is an accepted option for each node in the TreeView component.